classes/webservice/WebserviceSpecificManagementAttachments.php

```diff
@@ -0,0 +1,283 @@
+                    break;
+                case 'POST':
+                    $this->wsObject->executeEntityPost();
+                    break;
+                case 'PUT':
+                    $this->wsObject->executeEntityPut();
+                    break;
+                case 'DELETE':
+                    $this->wsObject->executeEntityDelete();
+                    break;
+            }
+        }
+        // Need to set an object for the WebserviceOutputBuilder object in any case
+        // because schema need to get webserviceParameters of this object
+        if (isset($object)) {
+            $this->wsObject->objects['empty'] = $object;
+        }
+    }
+
+    /**
+     * Handles attachment file download
+     *
+     * @throws WebserviceException if attachment is not existing or file not available
+     * @return string[] file details
+     */
+    public function executeFileGetAndHead()
+    {
+        $a = new Attachment((int) $this->wsObject->urlSegment[2]);
+        if ($a) {
+            // Physical file location
+            $file = _PS_DOWNLOAD_DIR_ . $a->file;
+            // Check if file exists
+            if (file_exists($file)) {
+                // Return file details
+                return array(
+                    'file' => $file,
+                    'mime' => $a->mime,
+                    'file_name' => $a->file_name,
+                    'file_size' => $a->file_size
+                );
+            } else {
+                throw new WebserviceException(sprintf('Unable to load the attachment file for attachment %d', $this->wsObject->urlSegment[2]), array(
+                    1,
+                    500,
+                ));
+            }
+        } else {
+            throw new WebserviceException(sprintf('Attachment %d not found', $this->wsObject->urlSegment[2]), array(
+                1,
+                500,
+            ));
+        }
+    }
+
+    /**
+     * Handles file upload
+     *
+     * Creates new attachment or replaces existing with a new file.
+     * [PUT] update existing attachment file
+     * [POST] create new attachment
+     */
+    public function executeFileAddAndEdit()
+    {
+        // Load attachment with or without id depending on method
+        $a = $this->wsObject->method == 'PUT' ? new Attachment((int) $this->wsObject->urlSegment[1]) : new Attachment();
+        
+        // Check form data
+        if (isset($_FILES['file']) && is_uploaded_file($_FILES['file']['tmp_name'])) {
+            
+            // Ensure file is within allowed size limit
+            if ($_FILES['file']['size'] > (Configuration::get('PS_ATTACHMENT_MAXIMUM_SIZE') * 1024 * 1024)) {
+                $this->wsObject->errors[] = sprintf($this->l('The file is too large. Maximum size allowed is: %1$d kB. The file you are trying to upload is %2$d kB.'), (Configuration::get('PS_ATTACHMENT_MAXIMUM_SIZE') * 1024), number_format(($_FILES['file']['size'] / 1024), 2, '.', ''));

```

Comment:
For the 1.6 branch, it is can be found in Crowdin in the 1.6/admin file.
For the 1.7.x branches, it is located in _Admin.Notifications.Error_.